### PR TITLE
TJT-1312 Add support for multiple providers in the object page

### PR DESCRIPTION
**Pull request by msaadq**
# Why?

To be able to see multiple shipping providers in the Object page when viewing listings that support Fiks Ferdig

# What?

`FiksFerdigShippingInfoView` has been modified to support multiple providers

# Version Change

It is a breaking change because the FINN app calls `FiksFerdigShippingInfoViewModel` which has been modified

# UI Changes

| Before | After |
| --- | --- |
|  Screenshot 2022-12-20 at 16 56 31 | Screenshot 2022-12-20 at 16 57 12 |

### CODE REVIEW

File: Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        let label = Label(style: .captionStrong, withAutoLayout: true)
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private let messageLabel: Label = {
+        let label = Label(style: .caption, withAutoLayout: true)
+        label.numberOfLines = 0
+        return label
+    }()
```
Comment (bstien) [inline Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift:17]: `numberOfLines` has been added as an init parameter, so these two can be shortened:

```swift
private let providerLabel = Label(style: .captionStrong, numberOfLines: 0, withAutoLayout: true)
private let messageLabel = Label(style: .caption, numberOfLines: 0, withAutoLayout: true)
```

File: Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        ])
+
+        self.addArrangedSubviews([
+            providerIconView,
+            textContainerStackView
+        ])
+    }
+}
+
+public class FiksFerdigShippingInfoCellViewModel {
```
Comment (AtishPapa) [inline Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift:72]: Please put this into a separate file. 😉

File: Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift (hunk 1 above)
Comment (msaadq) [reply Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift:17]: @bstien It doesn't show up in the latest version of the Finniverse SPM after I tried updating it. 
image

Should I change that SPM reference to point to `master` instead?

Comment (BjornRuud) [reply Sources/Components/Fiks Ferdig/FiksFerdigInfoView/FiksFerdigShippingInfoCell.swift:17]: No, don't use master unless in emergency. :D Check if FinniversKit is ready for a version bump and make a new release if that is what's required.

**Review by AtishPapa — APPROVED** [review]
Nice 👍

**Pull request merged**